### ISSUE TITLE
When stating a file, sometimes get a "Numerical result out of range" error

### ISSUE THREAD

**Issue by gregorygeller** [mar-file-system/marfs#152]
```
[root@cc-fta01 marfs]# pwd 
/campaign.gellergr/admins/gellergr/marfs
[root@cc-fta01 marfs]# for i in `seq 1 10` do; { ls; }
ls: cannot open directory .: Numerical result out of range
multi  packed  tree  unifile
ls: cannot open directory .: Numerical result out of range
multi  packed  tree  unifile
ls: cannot open directory .: Numerical result out of range
multi  packed  tree  unifile
multi  packed  tree  unifile
multi  packed  tree  unifile
multi  packed  tree  unifile
multi  packed  tree  unifile
multi  packed  tree  unifile
```

**Comment by gregorygeller** [comment]
This issue is breaking my test scripts.

**Comment by jti-lanl** [comment]
Is it always the first few consecutive tries that fail, like that?

**Comment by wfvining** [comment]
I am now able to reproduce this every time I ls.

**Comment by gregorygeller** [comment]
No...it seems random:


[root@cc-fta01 marfs]# for i in `seq 1 10` do; { ls; }

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

multi  packed  tree  unifile

ls: cannot open directory .: Numerical result out of range

**Comment by wfvining** [comment]
I believe the problem may be that the `rc_ssize` variable is cast to an int when it is returned from `opendir_md` and appears negative. I will make a change and test.

**Comment by wfvining** [comment]
Resolved: 937ea21

**Comment by brettkettering** [comment]
Nice catch, Will. People mock me when I say I like strictly typed languages, like Ada, but in Ada this tuoe of thing can't get past the compiler. :-)

Guess we could stipulate using LINT ti catch such things in C.

Brett

________________________________
